### DISCUSSION TITLE
Support for STM32WL55xx

### DISCUSSION THREAD

**Discussion by MarekNovakACRIOS** [Lora-net/LoRaMac-node#1036] [Q&A]
Dear community,

Is the support fo STM32WL55xx planned?
I checked the license and it should be possible to include it in /src/radio folder of the repo side by SX126x, SX127x support etc.

It will need some dependency cleanup from ST trace headers maybe, but otherwise it is integrablable I think.

I also modified portions in radio.c for this platform (RadioIrqProcess() and RadioOnDioIrq() - a lot of processing was done directly in ISR, although there is the IrqProcess entry in struct Radio_s, which should be polled from non-ISR context).

I attach my modified version in an archive and I am able to integrate it / create pull request if this is wanted.

Regards! 

[stm32WL55.zip](https://github.com/Lora-net/LoRaMac-node/files/5892823/stm32WL55.zip)

**Comment by mluis1** [comment]
At the moment we have no plans to add support for the STM32WL55xx platform.

We will consider this addition once we get access to a development platform.

In the meantime you can use the libraries provided by ST as the provided LoRaWAN stack is based on this project.
In discussions tab of this project there are already 2 topics talking about the same subject. (#1003 and #1023)

In the future it would be nice if you could post this kind of questions on the project Discussions tab. It is a better place to engage discussions and then we can agree if it is an issue or not.
If you don't mind I move this issue to the Discussions tab.

**Reply by MarekNovakACRIOS** [reply]
Thanks @mluis1 for letting me know abd moving to discussion - I was not aware of the discussion beeing used for such purposes. (I am more used to Gitlab and there is not such a thing).

Anyway, there is a development platform available for STM32WLE5CC/STM32WL55C - https://www.st.com/en/evaluation-tools/nucleo-wl55jc.html

I got mine from local ST distributor, but it should be also available commercialy I think. 

If someone finds it beneficial, I can share a proposal/pull request for the platform..

**Reply by mluis1** [reply]
Thanks for the proposition.

If you are still willing to create a pull-request please do it.
